addons/website_sale/models/product_template.py

```diff
@@ -196,7 +196,9 @@ def _get_sales_prices(self, pricelist):
 
             if template.compare_list_price:
                 # The base_price becomes the compare list price and the price_reduce becomes the price
-                base_price = template.compare_list_price
+                compare_list_price = self.currency_id._convert(template.compare_list_price, pricelist.currency_id, self.env.company,
+                                                  fields.Datetime.now(), round=False)
+                base_price = compare_list_price
                 if not price_list_contains_template:
                     price_reduce = base_sales_prices[template.id]
                     template_price_vals.update(price_reduce=price_reduce)
```

Comment:
Let's try to follow same code organisation as `master`
```suggestion
                base_price = template.compare_list_price
                if not price_list_contains_template:
                    price_reduce = base_sales_prices[template.id]

                if template.currency_id != pricelist.currency_id:
                    base_price = template.currency_id._convert(
                        base_price,
                        pricelist.currency_id,
                        self.env.company,
                        fields.Datetime.now(),
                        round=False
                    )
```
https://github.com/odoo/odoo/blob/ac3621084c9e4275f74d494cc193c09d2f9c3158/addons/website_sale/models/product_template.py#L202C1-L213C22